tests/unit/createBirthdaySquad.test.ts

```diff
@@ -0,0 +1,254 @@
+
+import * as getSquadMembers from "@/db/queries/getSquadMembers";
+import { squadJoins, users } from "@/db/schema";
+import type { Events } from "@/events";
+import { handler } from "@/functions/events/createBirthdaySquad";
+import { BIRTHDAY_SQUAD_SIZE } from "@/functions/utils/constants";
+import { openConversation } from "@/services/slack/openConversation";
+import { seedSquadJoins } from "@/testUtils/seedSquadJoins";
+import { testDb } from "@/testUtils/testDb";
+import { mockEventBridgePayload } from "@/testUtils/unit/mockEventBridgePayload";
+import { sendMockSqsMessage } from "@/testUtils/unit/sendMockSqsMessage";
+
+dayjs.extend(utc);
+
+const constants = vi.hoisted(() => ({
+  birthdayPerson: "U001",
+  eventId: "E001",
+  teamId: "T001",
+  otherUserIds: ["U002", "U003", "U004", "U005", "U006"],
+  converastionId: "CH001",
+}));
+
+vi.mock("@/services/slack/openConversation", async () => ({
+  openConversation: vi.fn().mockResolvedValue(constants.converastionId),
+}));
+
+describe("CreateBirthdaySquad", () => {
+  let openConversationMock: Mock;
+
+  beforeEach(() => {
+    openConversationMock = openConversation as Mock;
+  });
+
+  afterEach(async () => {
+    vi.clearAllMocks();
+  });
+
+  it("Should call getSquadMembers", async () => {
+    const event = {
+      birthdayPerson: constants.birthdayPerson,
+      eventId: constants.eventId,
+      team: constants.teamId,
+    } satisfies Events["createBirthdaySquad"];
+    const getSquadMembersSpy = vi.spyOn(getSquadMembers, "getSquadMembers");
+
+    openConversationMock.mockResolvedValueOnce(constants.converastionId);
+
+    await sendMockSqsMessage("createBirthdaySquad", event, handler);
+
+    expect(getSquadMembersSpy).toHaveBeenCalledWith(
+      constants.teamId,
+      constants.birthdayPerson,
+    );
+  });
+});
+
+describe("Final squad size is less then 2", () => {
+  let openConversationMock: Mock;
+  let eventBridge: EventBridgeClient;
+
+  beforeEach(() => {
+    openConversationMock = openConversation as Mock;
+    eventBridge = new EventBridgeClient();
+  });
+
+  afterEach(async () => {
+    vi.clearAllMocks();
+  });
+
+  it("Should not call openConverstaion", async () => {
+    const event = {
+      birthdayPerson: constants.birthdayPerson,
+      eventId: constants.eventId,
+      team: constants.teamId,
+    } satisfies Events["createBirthdaySquad"];
+    const getRandomSquadMembersSpy = vi.spyOn(
+      getSquadMembers,
+      "getRandomSquadMembers",
+    );
+
+    openConversationMock.mockResolvedValueOnce(constants.converastionId);
+
+    await sendMockSqsMessage("createBirthdaySquad", event, handler);
+
+    expect(getRandomSquadMembersSpy).toHaveBeenCalledWith({
+      teamId: constants.teamId,
+      usersToExclude: [constants.birthdayPerson],
+      limit: BIRTHDAY_SQUAD_SIZE,
+    });
+
+    expect(openConversation).not.toBeCalled();
+  });
+
+  it("Should not call publishEvent", async () => {
```

Comment:
To me this is not informative enough because I don't know what kind of event we were expecting.